homeassistant/components/rainmachine/binary_sensor.py

```diff
@@ -175,7 +175,7 @@ class ProvisionSettingsBinarySensor(RainMachineEntity, BinarySensorEntity):
     def update_from_latest_data(self) -> None:
         """Update the state."""
         if self.entity_description.key == TYPE_FLOW_SENSOR:
-            self._attr_is_on = self.coordinator.data["system"].get("useFlowSensor")
+            self._attr_is_on = self.coordinator.data.get("system").get("useFlowSensor")
```

Comment:
Correct me if I'm wrong, but this is likely to fail if `system` is not in `coordinator.data`. I think you should provide a default.
```suggestion
            self._attr_is_on = self.coordinator.data.get("system", {}).get("useFlowSensor")
```

Reply:
Good catch, thank you!